fili-core/src/test/groovy/com/yahoo/bard/webservice/table/availability/MetricUnionAvailabilitySpec.groovy

```diff
@@ -0,0 +1,233 @@
+    PhysicalTableSchema schema2
+
+    PhysicalTable physicalTable1
+    PhysicalTable physicalTable2
+
+    Set<PhysicalTable> physicalTables
+
+    MetricUnionAvailability metricUnionAvailability
+
+    def setup() {
+        availability1 = Mock(Availability)
+        availability2 = Mock(Availability)
+
+        availability1.getDataSourceNames() >> Sets.newHashSet(TableName.of('source1'))
+        availability2.getDataSourceNames() >> Sets.newHashSet(TableName.of('source2'))
+
+        metricColumn1 = new MetricColumn('metric1')
+        metricColumn2 = new MetricColumn('metric2')
+
+        schema1 = Mock(PhysicalTableSchema)
+        schema2 = Mock(PhysicalTableSchema)
+
+        physicalTable1 = Mock(PhysicalTable)
+        physicalTable2 = Mock(PhysicalTable)
+
+        physicalTable1.getAvailability() >> availability1
+        physicalTable2.getAvailability() >> availability2
+
+        physicalTable1.getSchema() >> schema1
+        physicalTable2.getSchema() >> schema2
+
+        physicalTables = [physicalTable1, physicalTable2] as Set
+    }
+
+    def "Metric columns are initialized by fetching columns from availabilities, not from physical tables"() {
+        given:
+        availability1.getAllAvailableIntervals() >> [
+                (metricColumn1): Collections.emptyList()
+        ]
+        availability2.getAllAvailableIntervals() >> [
+                (metricColumn2): Collections.emptyList()
+        ]
+
+        MetricColumn tableColumn1 = new MetricColumn("shouldNotBeReturned1")
+        MetricColumn tableColumn2 = new MetricColumn("shouldNotBeReturned2")
+
+        schema1.getColumns(_) >> Sets.newHashSet(tableColumn1)
+        schema2.getColumns(_) >> Sets.newHashSet(tableColumn2)
+
+        physicalTable1.getSchema() >> schema1
+        physicalTable2.getSchema() >> schema2
+
+        metricUnionAvailability = new MetricUnionAvailability(physicalTables, [metricColumn1, metricColumn2] as Set)
+
+        when:
+        Map<Availability, Set<MetricColumn>> availabilitiesToAvailableColumns = metricUnionAvailability.availabilitiesToAvailableColumns
+
+        then:
+        availabilitiesToAvailableColumns.size() == 2
+
+        availabilitiesToAvailableColumns.containsKey(availability1)
+        Set<MetricColumn> availableColumns1 = availabilitiesToAvailableColumns.get(availability1)
+        availableColumns1.contains(metricColumn1)
+        !availableColumns1.contains(tableColumn1)
+        !availableColumns1.contains(tableColumn2)
+
+        availabilitiesToAvailableColumns.containsKey(availability2)
+        Set<MetricColumn> availableColumns2 = availabilitiesToAvailableColumns.get(availability2)
+        availableColumns2.contains(metricColumn2)
+        !availableColumns2.contains(tableColumn1)
+        !availableColumns2.contains(tableColumn2)
+    }
+
+    @Unroll
+    def "checkDuplicateValue returns duplicates of metric column names from 2 physical tables in the case of #caseDescription"() {
```

Comment:
This description doesn't seem to make sense, since for many of the cases we're testing, there are no duplicates. Perhaps there's some more info from the `where` block that needs to be surfaced in the test name?